plugins/dell/fu-plugin-dell.c

```diff
@@ -278,7 +278,7 @@ fu_plugin_dock_node (FuPlugin *plugin, const gchar *platform,
 	}
 
 	dev = fu_device_new ();
-	fu_device_set_platform_id (dev, platform);
+	fu_device_set_physical_id (dev, platform);
 	dock_id = g_strdup_printf ("DELL-%s" G_GUINT64_FORMAT, component_guid);
```

Comment:
I wonder if for Dell plugin now we actually want this "dock_id" to be the logical ID.

Reply:
If you have a logical ID you also need to set a physical ID; although that's done for you if you're using FuUsbDevice.

Reply:
So previously Dell was doing set platform ID and then jam a GUID to the prefix of "DELL-" and set that as regular ID.
Now that a platform ID is being set, I'm wondering to drop the `fu_device_set_id` call and instead set that GUID to the logical ID.

Reply:
I think that makes sense, yes. Can you try it before I change it?

Reply:
Sure.

Reply:
Added two commits for it.  Ran into a problem that you weren't doing automagic ID generation for emitted devices too that I fixed.  feel free to squash if you want.